### PR TITLE
Add the README.md file for the `TodoTemplate` project (#1544)

### LINKED ISSUE #1544: `TodoTemplate` README file
We should provide a README.md file for the project template in which we introduce the template, its features, and the way customers can use it.

### PR DESCRIPTION
**Pull request by MeysamBahadori**
this closes #1544

### CODE REVIEW

File: src/Tooling/Bit.Tooling.Templates/TodoTemplate/Docs/FAQ.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# FAQ
+
+WE ARE HERE TO HELP
+
+## GENERAL
+
+
+
+## LICENSE
```
Comment (ysmoradi) [inline src/Tooling/Bit.Tooling.Templates/TodoTemplate/Docs/FAQ.md:9]: The repo has a license file. Please remove this part.

File: src/Tooling/Bit.Tooling.Templates/TodoTemplate/Docs/Getting-Started.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+## Development prerequisites
+
+- C# as the main development language.
+- [Asp.net core blazor](https://docs.microsoft.com/en-us/aspnet/core/blazor/?view=aspnetcore-6.0) as main development Back-End and Fron-End framework
+- [CSS ](https://www.google.com/url?sa=t&amp;rct=j&amp;q=&amp;esrc=s&amp;source=web&amp;cd=&amp;cad=rja&amp;uact=8&amp;ved=2ahUKEwji-KOu0pj4AhWwm_0HHeZQDzoQFnoECAgQAQ&amp;url=https%3A%2F%2Fwww.w3schools.com%2Fcss%2F&amp;usg=AOvVaw0Xtbw_GBAChsgvZNkPLVGb)&amp; [Sass ](https://www.google.com/url?sa=t&amp;rct=j&amp;q=&amp;esrc=s&amp;source=web&amp;cd=&amp;cad=rja&amp;uact=8&amp;ved=2ahUKEwjvgoO60pj4AhUCi_0HHVmXBMkQFnoECAgQAQ&amp;url=https%3A%2F%2Fsass-lang.com%2F&amp;usg=AOvVaw0p_IRgLEbIPRGWtlW7Wph8)as stylesheet
+- [Ef Core](https://docs.microsoft.com/en-us/ef/core/) as ORM to communicate with the database
+- [Asp.Net Identity](https://docs.microsoft.com/en-us/aspnet/identity/overview/getting-started/introduction-to-aspnet-identity) with [JWT ](https://www.c-sharpcorner.com/article/jwt-authentication-and-authorization-in-net-6-0-with-identity-framework/)supporting for handling Authentication
+
+## Pre Requirements tools
```
Comment (ysmoradi) [inline src/Tooling/Bit.Tooling.Templates/TodoTemplate/Docs/Getting-Started.md:13]: Rename it to environment setup

File: src/Tooling/Bit.Tooling.Templates/TodoTemplate/Docs/Getting-Started.md [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+- Microsoft Visual Studio 2022 - Preview Version 17.3.0 Preview 1.0 or higher
```
Comment (ysmoradi) [inline src/Tooling/Bit.Tooling.Templates/TodoTemplate/Docs/Getting-Started.md:15]: VS 2022 17.3 ..... with the following workloads:
1- ASP.NET and web...
2- .NET Multi...
and the following extensions:
1- Web Compiler 2022+
installed.

File: src/Tooling/Bit.Tooling.Templates/TodoTemplate/Docs/Getting-Started.md [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+- Asp.net and web development VS workload
+- .Net Multi-Platform App UI development (for mobile app development target) VS workload
+
+## Setup envirnoment
```
Comment (ysmoradi) [inline src/Tooling/Bit.Tooling.Templates/TodoTemplate/Docs/Getting-Started.md:19]: Remove this section

File: src/Tooling/Bit.Tooling.Templates/TodoTemplate/Docs/Getting-Started.md [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+vs extention...
+
+
+## Create project
+
+Clone the [TodoTemplate](https://github.com/bitfoundation/bitplatform/tree/develop/src/Tooling/Bit.Tooling.Templates/TodoTemplate) from the Bit-Platform Github repository
```
Comment (ysmoradi) [inline src/Tooling/Bit.Tooling.Templates/TodoTemplate/Docs/Getting-Started.md:25]: Use dotnet new approach based on assumption that we've `Bit.Tooling.Templates.TodoTemplate` nuget package

File: src/Tooling/Bit.Tooling.Templates/TodoTemplate/Docs/Getting-Started.md [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+
+# Configure The Project
+
+## Restore NuGet packages
```
Comment (ysmoradi) [inline src/Tooling/Bit.Tooling.Templates/TodoTemplate/Docs/Getting-Started.md:29]: Remove this section

File: src/Tooling/Bit.Tooling.Templates/TodoTemplate/Docs/Getting-Started.md [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+
+If automatically not started click on the &quot;Restore NuGet packages&quot; item from right-click menu of solution name in solution explorer windows.
+
+## Database
+
+**Connection String**
+
+Open  **appsettings.json**  in  **TodoTemplate.Api**  project and change the  **SqlServerConnection ** connection string if you want:
+
+&quot;ConnectionStrings&quot;: {
```
Comment (ysmoradi) [inline src/Tooling/Bit.Tooling.Templates/TodoTemplate/Docs/Getting-Started.md:39]: Formatting issue

File: src/Tooling/Bit.Tooling.Templates/TodoTemplate/Docs/Getting-Started.md [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+
+You can use Entity Framework&#39;s built-in tools for migrations. Open  **Package Manager Console**  in Visual Studio set  **TodoTemplate.Api**  as the  **Default Project**  and run the  **Update-Database**  command as shown below:
+
+    Update-Database -Context TodoTemplateDbContext
+
+2. **Migrator Application**
+
+Bit-Platform includes a **TodoTemplate.Migrator**  project in the solution. You can run this tool for database migrations on development and production platforms.
+
+All things you need is to set this project as a startup in solution and run it.
```
Comment (ysmoradi) [inline src/Tooling/Bit.Tooling.Templates/TodoTemplate/Docs/Getting-Started.md:59]: Let's forget about Migrator project at this time

File: src/Tooling/Bit.Tooling.Templates/TodoTemplate/Docs/Getting-Started.md [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+
+## BlazorMode
+Bit-Platform support tree modes for Blazor include BlazorServer, BlazorWebAssembly, and BlazorHybrid which can easily switch between it.
+
+### BlazorServer
```
Comment (ysmoradi) [inline src/Tooling/Bit.Tooling.Templates/TodoTemplate/Docs/Getting-Started.md:64]: Please provide instructions. Also, you can talk a little about why the blazor server matters?

File: src/Tooling/Bit.Tooling.Templates/TodoTemplate/Docs/Getting-Started.md [hunk 10]
```diff
@@ -0,0 +1,78 @@
...
+
+### BlazorWebAssembly
```
Comment (ysmoradi) [inline src/Tooling/Bit.Tooling.Templates/TodoTemplate/Docs/Getting-Started.md:66]: The same for blazor web assembly

File: src/Tooling/Bit.Tooling.Templates/TodoTemplate/Docs/Getting-Started.md [hunk 11]
```diff
@@ -0,0 +1,78 @@
...
+
+### BlazorHybrid
+
```
Comment (ysmoradi) [inline src/Tooling/Bit.Tooling.Templates/TodoTemplate/Docs/Getting-Started.md:69]: The same for blazor hybrid

File: src/Tooling/Bit.Tooling.Templates/TodoTemplate/Docs/Getting-Started.md [hunk 12]
```diff
@@ -0,0 +1,78 @@
...
+## Run
```
Comment (ysmoradi) [inline src/Tooling/Bit.Tooling.Templates/TodoTemplate/Docs/Getting-Started.md:70]: Do we need a separated run section? We've discussed how to run the app in 3 different blazor modes before

File: src/Tooling/Bit.Tooling.Templates/TodoTemplate/Docs/Getting-Started.md [hunk 13]
```diff
@@ -0,0 +1,78 @@
...
+How run...
+
+**Note**: wilcard & admin vs ...
```
Comment (ysmoradi) [inline src/Tooling/Bit.Tooling.Templates/TodoTemplate/Docs/Getting-Started.md:73]: You've to make it clear that vs needs to be started using administrator privileges for mobile development in blazor hybrid section

File: src/Tooling/Bit.Tooling.Templates/TodoTemplate/Docs/Introduction.md [hunk 14]
```diff
@@ -0,0 +1,37 @@
+# Introduction
```
Comment (ysmoradi) [inline src/Tooling/Bit.Tooling.Templates/TodoTemplate/Docs/Introduction.md:1]: These contents are redundant. We'll have contents like these in readme.md in the repository's root folder and in bitplatform.dev site

File: src/Tooling/Bit.Tooling.Templates/TodoTemplate/Docs/Publish.md [hunk 15]
```diff
@@ -0,0 +1,6 @@
+# Deployment & Publish
+
+## Publishing to IIS
```
Comment (ysmoradi) [inline src/Tooling/Bit.Tooling.Templates/TodoTemplate/Docs/Publish.md:3]: Let's talk about these steps:
Switch to either blazor hybrid / web assembly
Set deployment type to either `default`, `ssr` or `pwa`
We've nothing to do with IIS here.

**Review by ysmoradi — CHANGES_REQUESTED** [review]

**Comment by ysmoradi** [comment]
How to change email template contents and test them (Local folder concept)? How to use the email server in prod?
How to change the password policy for signup?
What are known and unknown exceptions?
Talk about `TodoTemplateJsonContext` and System.Text.Json source generator
How the mechanism of syncing code between app and web work? (Talk about duplicated files such as index.html)
Client side exception handling `WrapHandled`

File: src/Tooling/Bit.Tooling.Templates/TodoTemplate/Docs/Getting-Started.md (hunk 2 above)
Comment (MeysamBahadori) [reply src/Tooling/Bit.Tooling.Templates/TodoTemplate/Docs/Getting-Started.md]: OK

File: src/Tooling/Bit.Tooling.Templates/TodoTemplate/Docs/Getting-Started.md (hunk 3 above)
Comment (MeysamBahadori) [reply src/Tooling/Bit.Tooling.Templates/TodoTemplate/Docs/Getting-Started.md]: Ok

File: src/Tooling/Bit.Tooling.Templates/TodoTemplate/Docs/Getting-Started.md (hunk 4 above)
Comment (MeysamBahadori) [reply src/Tooling/Bit.Tooling.Templates/TodoTemplate/Docs/Getting-Started.md]: Removed

File: src/Tooling/Bit.Tooling.Templates/TodoTemplate/Docs/Getting-Started.md (hunk 5 above)
Comment (MeysamBahadori) [reply src/Tooling/Bit.Tooling.Templates/TodoTemplate/Docs/Getting-Started.md]: Ok added

File: src/Tooling/Bit.Tooling.Templates/TodoTemplate/Docs/Getting-Started.md (hunk 6 above)
Comment (MeysamBahadori) [reply src/Tooling/Bit.Tooling.Templates/TodoTemplate/Docs/Getting-Started.md]: Removed

File: src/Tooling/Bit.Tooling.Templates/TodoTemplate/Docs/Getting-Started.md (hunk 7 above)
Comment (MeysamBahadori) [reply src/Tooling/Bit.Tooling.Templates/TodoTemplate/Docs/Getting-Started.md]: Ok

File: src/Tooling/Bit.Tooling.Templates/TodoTemplate/Docs/Getting-Started.md (hunk 9 above)
Comment (MeysamBahadori) [reply src/Tooling/Bit.Tooling.Templates/TodoTemplate/Docs/Getting-Started.md:55]: Added and edited

File: src/Tooling/Bit.Tooling.Templates/TodoTemplate/Docs/Getting-Started.md (hunk 10 above)
Comment (MeysamBahadori) [reply src/Tooling/Bit.Tooling.Templates/TodoTemplate/Docs/Getting-Started.md:59]: Added and edited

File: src/Tooling/Bit.Tooling.Templates/TodoTemplate/Docs/Getting-Started.md (hunk 11 above)
Comment (MeysamBahadori) [reply src/Tooling/Bit.Tooling.Templates/TodoTemplate/Docs/Getting-Started.md:66]: Added and edited

File: src/Tooling/Bit.Tooling.Templates/TodoTemplate/Docs/Getting-Started.md (hunk 12 above)
Comment (MeysamBahadori) [reply src/Tooling/Bit.Tooling.Templates/TodoTemplate/Docs/Getting-Started.md]: Ok,  was corrected

File: src/Tooling/Bit.Tooling.Templates/TodoTemplate/Docs/Getting-Started.md (hunk 13 above)
Comment (MeysamBahadori) [reply src/Tooling/Bit.Tooling.Templates/TodoTemplate/Docs/Getting-Started.md]: added

File: src/Tooling/Bit.Tooling.Templates/TodoTemplate/Docs/Introduction.md (hunk 14 above)
Comment (MeysamBahadori) [reply src/Tooling/Bit.Tooling.Templates/TodoTemplate/Docs/Introduction.md]: Entirely removed

File: src/Tooling/Bit.Tooling.Templates/TodoTemplate/Docs/Publish.md (hunk 15 above)
Comment (MeysamBahadori) [reply src/Tooling/Bit.Tooling.Templates/TodoTemplate/Docs/Publish.md]: Ok, Added and remove iis

**Comment by msynk** [comment]
what is the issue of this PR?

**Comment by ysmoradi** [comment]
> what is the issue of this PR?

#1544

File: src/Tooling/Bit.Tooling.Templates/TodoTemplate/Directory.build.props [hunk 16]
```diff
@@ -1,7 +1,7 @@
 <Project>
 
     <PropertyGroup>
-        <BlazorMode>BlazorServer</BlazorMode>
+        <BlazorMode>BlazorWebAssembly</BlazorMode>
```
Comment (ysmoradi) [inline src/Tooling/Bit.Tooling.Templates/TodoTemplate/Directory.build.props:4]: Please revert this change

Comment (MeysamBahadori) [reply src/Tooling/Bit.Tooling.Templates/TodoTemplate/Directory.build.props]: Ok

**Comment by msynk** [comment]
@MeysamBahadori 
@ysmoradi 
why this PR is closed?

**Comment by MeysamBahadori** [comment]
This pr was based on the Develop Branch, which prevented parallel development, so it was removed and built on the related Issue branch base.

 Senior .NET (Full-stack) Engineer
   Meysam bahadori (+989397524270)
________________________________
